internal/contract/controlplane.go

```diff
@@ -101,53 +101,86 @@ func (c *ControlPlaneContract) StatusReplicas() *Int64 {
 }
 
 // UpdatedReplicas provide access to the status.updatedReplicas field in a ControlPlane object, if any. Applies to implementations using replicas.
-func (c *ControlPlaneContract) UpdatedReplicas() *Int64 {
+// TODO (v1beta2): Rename to V1Beta1DeprecatedUpdatedReplicas and make sure we are only using this method for compatibility with old contracts.
+func (c *ControlPlaneContract) UpdatedReplicas(contractVersion string) *Int64 {
```

Comment:
Not sure I understand how this fixes the test. Wasn't KCP still declaring the v1beta1 contract?